File: MTGSearch/src/androidTest/java/com/dbottillo/mtgsearchfree/dagger/DataModuleForTest.java
package com.dbottillo.mtgsearchfree.dagger;

import android.content.Context;
import android.content.SharedPreferences;

import com.dbottillo.mtgsearchfree.model.storage.CardsPreferences;
import com.dbottillo.mtgsearchfree.model.storage.CardsPreferencesForTest;
import com.dbottillo.mtgsearchfree.model.storage.GeneralData;
import com.dbottillo.mtgsearchfree.model.storage.GeneralPreferencesForTest;
import com.dbottillo.mtgsearchfree.util.AppInfo;

import javax.inject.Singleton;

import dagger.Provides;

public class DataModuleForTest extends DataModule {

    @Provides
    @Singleton
    GeneralData providesGeneralData(SharedPreferences sharedPreferences, AppInfo appInfo) {
        return new GeneralPreferencesForTest();
    }

    @Provides
    @Singleton
    CardsPreferences provideGeneralPreferences(Context context) {
        return new CardsPreferencesForTest(context);
    }
}

File: app/src/test/kotlin/com/dbottillo/mtgsearchfree/dagger/DataModuleForTest.kt
package com.dbottillo.mtgsearchfree.dagger

import android.content.Context
import com.dbottillo.mtgsearchfree.model.storage.CardsPreferences
import com.dbottillo.mtgsearchfree.model.storage.CardsPreferencesForTest
import com.dbottillo.mtgsearchfree.model.storage.GeneralData
import com.dbottillo.mtgsearchfree.model.storage.GeneralPreferencesForTest
import com.dbottillo.mtgsearchfree.util.AppInfo
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class DataModuleForTest : DataModule() {

    @Provides
    @Singleton
    override fun providesGeneralData(context: Context, appInfo: AppInfo): GeneralData {
        return GeneralPreferencesForTest()
    }

    @Provides
    @Singleton
    override fun provideGeneralPreferences(context: Context): CardsPreferences {
        return CardsPreferencesForTest(context)
    }
}
